seatunnel-connectors-v2/connector-email/src/main/java/org/apache/seatunnel/connectors/seatunnel/email/sink/EmailSinkWriter.java

```diff
@@ -0,0 +1,158 @@
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.email.sink;
+
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.connectors.seatunnel.common.sink.AbstractSinkWriter;
+import org.apache.seatunnel.connectors.seatunnel.email.config.EmailSinkConfig;
+
+import org.apache.seatunnel.shade.com.typesafe.config.Config;
+
+import com.sun.mail.util.MailSSLSocketFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.activation.FileDataSource;
+import javax.mail.Authenticator;
+import javax.mail.BodyPart;
+import javax.mail.Message;
+import javax.mail.Multipart;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Properties;
+
+public class EmailSinkWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EmailSinkWriter.class);
+
+    private final SeaTunnelRowType seaTunnelRowType;
+    private EmailSinkConfig config;
+    private StringBuffer stringBuffer;
+
+    public EmailSinkWriter(SeaTunnelRowType seaTunnelRowType, Config pluginConfig) {
+        this.seaTunnelRowType = seaTunnelRowType;
+        this.config = new EmailSinkConfig(pluginConfig);
+        this.stringBuffer = new StringBuffer();
+    }
+
+    @Override
+    public void write(SeaTunnelRow element) {
+        Object[] fields = element.getFields();
+
+        for (Object field : fields) {
+            stringBuffer.append(field.toString() + ",");
+        }
+        stringBuffer.deleteCharAt(fields.length - 1);
+        stringBuffer.append("\n");
+
+    }
+
+    @Override
+    public void close() {
+        createFile();
+        Properties properties = new Properties();
+
+        properties.setProperty("mail.host", "smtp.qq.com");
+
+        properties.setProperty("mail.transport.protocol", "smtp");
+
+        properties.setProperty("mail.smtp.auth", "true");
```

Comment:
Same problem as above, and these methods may not be appropriate in `close` method